### PR TITLE
Simulated Data + Plots

### PR DESCRIPTION
**Pull request by murphymadeleine21**
First of probably a few updates for the simulated data code.

### CODE REVIEW

**Comment by murphymadeleine21** [comment]
So, here are a few results that confuse me a bit. Below is the results we get from basically running figure 1 on the simulated data (now a new Figure4). It takes the simulated 4D continuous dataset, plus that 4D tensor shrunk into 3D, and runs the various factorizations a bunch of times. 
Screen Shot 2022-03-10 at 4 01 48 PM
In the first subplot we can see a bit of blue coming out from under the orange which is good. But in B we only see the blue points, and I think that the green and orange ones should be visible given that they have different factorization sizes?

I had the function print out the R2X values just to see, and it looks like they _are_ all very similar, so it makes sense why we might have some of them be overlaid.
Screen Shot 2022-03-10 at 4 01 56 PM

But looking at original figure 1, the sizes should be different? I guess I should maybe have the function print out sizes as well just to be sure, because otherwise I don't know why we don't see distinct points in Fig4.
Fig 1:
Screen Shot 2022-03-10 at 4 02 01 PM

File: syserol/simulated.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    random_cp.time = dayLabels()
+    # Generate P matrix with realistic values.
+    # Based on curve simulation, a, b, c, and d can/will be in the ranges below
+    P = np.zeros((4, r))
+    for i in range(r):
+        a = np.random.uniform(0.01, 1)
+        b = np.random.uniform(0.01, 1.2)
+        c = np.random.uniform(0.01, 30)
+        d = np.random.uniform(0.01, 100)
+        P[:, i] = a, b, c, d
```
Comment (murphymadeleine21) [inline syserol/simulated.py:30]: Also, here you can see what I've done for the P matrix for now, which is _semi_-manual. I have set these manual parameter ranges that it is currently randomly selecting from, but Cyrillus and I talked yesterday about picking a specific set of these that will generate different looking curves for before&after factorization analysis.

**Comment by cyrillustan** [comment]
1. In fig. 4: (a) maybe zoom into a higher range of R2X (not starting from 0.0, but maybe 0.8?). Currently, it's hard to say anything; (b) Why are there only blue points?
2. Fig 5: Hmm, interesting, definitely worth looking into more. First, try to be consistent with direction and scale. If the left-hand plot is 0-1 but allows decreasing curves, do so for the right-hand side plot too.

**Comment by aarmey** [comment]
Yes, the sizes should be different. Maybe print those out to check?

**Comment by murphymadeleine21** [comment]
@cyrillustan Thanks for the tips and I agree about figure 5, I enforced the same constraints we use in `reorient_factors` which handles what we see in the right figure (the time components after factorization). I think this should do the trick?
But still, we see a difference here
Screen Shot 2022-03-11 at 4 15 49 PM

**Comment by murphymadeleine21** [comment]
Screen Shot 2022-04-06 at 1 45 10 PM
@cyrillustan @aarmey 
This actually looks pretty good. I designed 4 specific curves that I think have pretty distinct shapes. After removing 50% of the data, here is an example of one output figure. It does change from run to run (because different values are removed) but this is one example of a good run. The only thing is that the component order is different -- I actually don't understand this because I run reorient/sort factors on the simulated data. But to me it looks pretty clear that the factorization is capturing the right patterns, just the component numbers are different.

**Comment by murphymadeleine21** [comment]
Screen Shot 2022-04-06 at 1 52 22 PM
Figure 4 on the other hand is kind of ugly because of the vast difference in factorization size. Not sure if there's a better way to view it

**Comment by murphymadeleine21** [comment]
Ok. For now I will change it to 4 components and then look into possibly changing the number of timepoints. Right now we just have it building exactly the same as the Zohar data set (we call `build_cFactor` on the P matrix).

Also, I wanted to see what the factorization looked like when adding some noise too.
Screen Shot 2022-04-07 at 12 10 55 PM

Thought this run looked pretty good. The third plot is after adding normally distributed noise to the original simulated data set. Probably need to set a random seed to make the runs consistent? Also, another question I had was how do we add "more" or "less" noise? Would that just mean the number of points in the data set we're actually adding noise to? In this example, I add noise to every data point.

**Comment by aarmey** [comment]
Looks great! "More" noise would mean still adding noise to every point, but increasing the magnitude of the noise term. For example, if you're adding a randn vector, multiply it by some number before adding it in.

**Pull request merged**